torch/optim/adam.py

```diff
@@ -305,7 +315,7 @@ def _single_tensor_adam(params: List[Tensor],
 
             if amsgrad:
                 # Maintains the maximum of all 2nd moment running avg. till now
-                torch.maximum(max_exp_avg_sqs[i], exp_avg_sq, out=max_exp_avg_sqs[i])
+                torch.maximum(max_exp_avg_sqs[i].clone(), exp_avg_sq, out=max_exp_avg_sqs[i])
```

Comment:
Only if differentiable?

Reply:
this was not needed, the differentiable case is now handled in the outer if :)